torchreid/engine/engine.py

```diff
@@ -142,7 +147,13 @@ def run(
                     use_metric_cuhk03=use_metric_cuhk03,
                     ranks=ranks
                 )
-                self._save_checkpoint(epoch, rank1, save_dir)
+                if not save_best_only:
+                    self._save_checkpoint(epoch, rank1, save_dir)
+                # save best epoch on test set
+                elif rank1 >= rank1_best:
+                    rank1_best = rank1
+                    self._save_checkpoint(epoch, rank1, save_dir, is_best=True)
```

Comment:
we also need to modify `save_checkpoint()` in https://github.com/KaiyangZhou/deep-person-reid/blob/66df1060166b2fa44dc5e45460b649b9ddcbd827/torchreid/utils/torchtools.py#L20, otherwise the intermediate checkpoints will still be saved